flytekit/annotated/task.py

```diff
@@ -433,6 +469,55 @@ def execute(self, **kwargs) -> Any:
                 logger.info("Executing Dynamic workflow, using raw inputs")
                 return self._task_function(**kwargs)
 
+        if ctx.execution_state and ctx.execution_state.mode == ExecutionState.Mode.TASK_EXECUTION:
+            return self.compile_into_workflow(ctx, **kwargs)
+
+    def dispatch_execute(
+        self, ctx: FlyteContext, input_literal_map: _literal_models.LiteralMap
+    ) -> _literal_models.LiteralMap:
+        """
+        This is a copy of the base dispatch_execute, except that we just return the dj spec generated when running
+        for real.
+        """
+
+        # Translate the input literals to Python native
+        native_inputs = flytekit_engine.idl_literal_map_to_python_value(ctx, input_literal_map)
+
+        # TODO: Logger should auto inject the current context information to indicate if the task is running within
+        #   a workflow or a subworkflow etc
+        logger.info(f"Invoking {self.name} with inputs: {native_inputs}")
+        try:
+            native_outputs = self.execute(**native_inputs)
+        except Exception as e:
+            logger.exception("Exception when executing", e)
+            raise e
+        logger.info(f"Task executed successfully in user level, outputs: {native_outputs}")
+
+        # Short circuit the translation to literal map cuz what's returned is a dj spec (or an already-constructed
+        # LiteralMap if the dynamic task was a no-op), not python native values
+        if ctx.execution_state and ctx.execution_state.mode == ExecutionState.Mode.TASK_EXECUTION:
```

Comment:
so basically you want to modify the dispatch execute to return dynamicjobspec if that exists or return literalMap. Why cant we check for the djspec type?

Reply:
done.